src/DI/ApiClientsExtension.php

```diff
@@ -17,65 +27,62 @@
 class ApiClientsExtension extends CompilerExtension
 {
 
-	/** @var AbstractPass[] */
-	protected $passes = [];
-
-	/** @var string[] */
-	protected $map = [
-		AppLotusPass::APP_NAME => AppLotusPass::class,
-	];
-
 	public function getConfigSchema(): Schema
 	{
 		return Expect::structure([
 			'debug' => Expect::bool(false),
-			'app' => Expect::structure([
-				AppLotusPass::APP_NAME => Expect::anyOf(null, AppLotusPass::getConfigSchema()),
-			])->castTo('array'),
+			'http' => Expect::structure([
+				'http_errors' => Expect::bool(false),
+				'base_uri' => Expect::string()->required(),
+				'headers' => Expect::structure([
+					'X-Api-Token' => Expect::string()->required(),
+				])->castTo('array'),
+			])->otherItems(Expect::mixed())
+				->castTo('array'),
 		])->castTo('array');
 	}
 
-	public function __construct()
-	{
-		$this->passes[] = new CorePass($this);
-	}
-
 	public function loadConfiguration(): void
 	{
-		$config = $this->config;
+		$config = $this->getConfig();
+		$builder = $this->getContainerBuilder();
 
-		// Instantiate and configure enabled passes
-		foreach ($this->map as $passName => $passClass) {
-			$passConfig = $config['app'][$passName];
-			if ($passConfig === null) {
-				continue;
-			}
+		$builder->addDefinition($this->prefix('guzzleFactory'))
+			->setFactory(GuzzleFactory::class);
 
-			/** @var AbstractPass $pass */
-			$this->passes[] = $pass = new $passClass($this);
-			$pass->setConfig($passConfig);
-		}
+		// #1 HTTP client
+		$builder->addDefinition($this->prefix('http.client'))
+			->setFactory($this->prefix('@guzzleFactory::create'), [$config['http']])
+			->setType(HttpClient::class)
+			->setAutowired(false);
 
-		// Trigger passes
-		foreach ($this->passes as $pass) {
-			$pass->loadPassConfiguration();
-		}
-	}
+		// #2 Clients
+		$builder->addDefinition($this->prefix('client.calendar'))
+			->setFactory(CalendarClient::class, [$this->prefix('@http.client')]);
+		$builder->addDefinition($this->prefix('client.plan'))
+			->setFactory(PlanClient::class, [$this->prefix('@http.client')]);
+		$builder->addDefinition($this->prefix('client.process'))
+			->setFactory(ProcessClient::class, [$this->prefix('@http.client')]);
+		$builder->addDefinition($this->prefix('client.snippet'))
+			->setFactory(SnippetClient::class, [$this->prefix('@http.client')]);
+		$builder->addDefinition($this->prefix('client.user'))
+			->setFactory(UserClient::class, [$this->prefix('@http.client')]);
+		$builder->addDefinition($this->prefix('client.userGroup'))
+			->setFactory(UserGroupClient::class, [$this->prefix('@http.client')]);
 
-	public function beforeCompile(): void
-	{
-		// Trigger passes
-		foreach ($this->passes as $pass) {
-			$pass->beforePassCompile();
-		}
-	}
-
-	public function afterCompile(ClassType $class): void
-	{
-		// Trigger passes
-		foreach ($this->passes as $pass) {
-			$pass->afterPassCompile($class);
-		}
+		// #3 Requestors
+		$builder->addDefinition($this->prefix('requestor.calendar'))
```

Comment:
there is still an open posibility to merge clients and services and return LotusResponse which container both data and psr